build.gradle

```diff
@@ -35,6 +35,9 @@ dependencies {
 	implementation 'org.jsoup:jsoup:1.13.1'
 	implementation 'org.codejive:java-properties:0.0.3'
 
+	implementation 'org.fusesource.jansi:jansi:2.4.0'
```

Comment:
This adds 1.1MB in download - 15% increase in size *just* to enable colors on special case Windows terminal feels overkill to me ;/

is it not just on mingw /msys which are old? basically every other setup works?

Reply:
Yes, that is very unfortunate. I have however not seen any Windows setup where colors do work, except for `git-bash.exe`, but I don't care much for that terminal. Can you tell me in which Windows aetup this should work? AFAIK I am using the latest version of all tools.

Reply:
> This adds 1.1MB in download

Hmm... This seems a bit odd. The library is only supposed to be 215 KB in size and has no dependencies on its own. I will take another look at that then.

Reply:
I compared the size of the latest `jbang.zip` and it looks like the Jansi dependency adds another 203 KB (~3.6%). I don't like it either, but I am not aware of any alternative (it looks like Quarkus' dev mode Maven mojo also relies on the Jansi bundled with Maven).

I would however still be curious on how I can get colors on Windows without adding Jansi (and not using `git-bash.exe`). I also tried some other terminals (like wezterm), but the result was always the same: ANSI sequences printed to the console.

Reply:
Thanks for the info.

> lower left: cygwin - bash, colors

I didn't try Cygwin. I don't know if anybody still uses that these days.

> lower right: windows terminal running wsl with ubuntu, colors

Yes, in Linux under WSL it all works nicely.

> Thus mere lack of color I'm not willing to add 15% "weight" for.

I can understand that. But AFAICT the size increase is not 1.1 MB but more like 203 KiB, which may still be acceptable.

> curious to know what situatons you are seeing garbled output and if you possibly be running 2+ year old windows?

No, I am using a recent Windows 10 build (19044/21H2 which is the build from November 2021) and also the most recent Windows Terminal (Terminal Preview 1.13.10395.0 and Terminal 1.12.10393.0). The difference is that I am running Git's `bash.exe` inside Windows Terminal. I think this is a rather common setup these days for us poor Windows users (I think the Git installer will optionally even create this profile for you) and this is where I have the problems with the ANSI codes.

Reply:
BTW if I set `TERM=xterm` in Windows Terminal with Git Bash then the colors come out fine. Not sure who's wrong here, the picocli terminal detection or WT/Git Bash setting the TERM varaible.

Reply:
BTW2: If I set `CLICOLOR_FORCE=1` for powershell and CMD I get colors as well, both in Windows Terminal and in their "original" windows. So it seems the colors work jsut fine, it's just picocli deciding not to use them?

Reply:
ANSI support on Windows is just a mess. If it can be fixed by setting an environment variable, then the fix will be much simpler and leaner.

Reply:
BTW3: I just noticed/remembered that for Linux/Mac we actually always force Picocli to use colors, bypassing its detection mechanism, which means we always output colors even when the user redirects the output to a file. That is kind of heavy-handed, but perhaps we should just do the same for Windows?

Reply:
Do please list what terminal and shell you are running when seeing these issues ?

Reply:
> Do please list what terminal and shell you are running when seeing these issues ?

I did: "Windows Terminal with Git Bash" :-)

Reply:
I have now also tested JBang on my private Windows laptop and there it of course just works... Possibly it has something to do with the Git installation. The installer offers a few options which seem like they could make a difference (see screenshots), but I have now tested many permutations of options (including some obscure environment variables and registry entries) and haven't been able to figure out what causes it to work or not work.

I think it is probably not worth investing more time in this. Maybe someone else who knows more about all this will in the future be able to figure out what is what so that it can be properly fixed or documented (would also be nice for picocli claiming that Jansi is required).

![image](https://user-images.githubusercontent.com/408765/156243157-1ce58afd-8cba-4376-8905-2b25cba09372.png)
![image](https://user-images.githubusercontent.com/408765/156243247-76a5e850-9dd9-4920-9483-9ed16777534a.png)